specification/datafactory/resource-manager/Microsoft.DataFactory/preview/2017-09-01-preview/entityTypes/LinkedService.json

```diff
@@ -95,6 +95,18 @@
           "description": "The connection string.",
           "$ref": "../datafactory.json#/definitions/SecretBase"
         },
+        "servicePrincipalId": {
+          "type": "object",
+          "description": "The ID of the service principal used to authenticate against Azure SQL Data Warehouse. Type: string (or Expression with resultType string)."
+        },
+        "servicePrincipalKey": {
+          "$ref": "../datafactory.json#/definitions/SecretBase",
+          "description": "The key of the service principal used to authenticate against Azure SQL Data Warehouse."
+        },
+        "tenant": {
+          "type": "object",
```

Comment:
We've been talking about this a lot lately. It seems like we actually don't have a great way to model this ATM. [Technically, a `string` is not a JSON `object`](json.org). However, the ability to use `oneOf` in your Swagger won't be available until we begin supporting Swagger v3 specification. There's a more thorough conversation happening here: https://github.com/Azure/autorest/issues/2827

At the moment, you're still doing the right thing, but I want to spread awareness of this problem.